### PR TITLE
Connection pool many connections fix

### PR DESCRIPTION
**Pull request by neunhoef**
This fixes a bug in the connectionPool implementation, which could
lead to the fact that basically no connections have been reused.

Under high load and high parallelism it could happen that a connection
was leased from fuerte but could then not be leased in the connection
pool. In this case the lease from fuerte was not returned and thus the
connection could no longer be used. In k8s and with TLS we have observed
situations where the timing was so that this built up and created
thousands of connections in a very short period of time.

This let performance suffer seriously.

The solution is essentially to release the lease immediately in the
above mentioned case.

We also added some more metrics to make this behaviour better
observable.

## Acceptance Checklist

*The below list is not exhaustive, think thoroughly whether the provided information is sufficient.*
*Remove options that do not apply*

### Scope & Purpose

*(Can you describe what functional change your PR is trying to effect?)*

- [*] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [ ] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)

### Testing & Verification

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13126/

**Review by jsteemann — COMMENTED** [review]
IMO this PR depends on https://github.com/arangodb/arangodb/pull/13057 being merged, so we should merge these two together.

File: arangod/Network/ConnectionPool.cpp [hunk 1]
```diff
@@ -196,6 +214,10 @@ std::shared_ptr<fuerte::Connection> ConnectionPool::createConnection(fuerte::Con
 
 ConnectionPtr ConnectionPool::selectConnection(std::string const& endpoint,
                                                ConnectionPool::Bucket& bucket) {
+  using namespace std::chrono;
+
+  auto start = high_resolution_clock::now();
```
Comment (jsteemann) [inline arangod/Network/ConnectionPool.cpp:219]: from https://en.cppreference.com/w/cpp/chrono/high_resolution_clock:

> The high_resolution_clock is not implemented consistently across different standard library implementations, and its use should be avoided. It is often just an alias for std::chrono::steady_clock or std::chrono::system_clock, but which one it is depends on the library or configuration. When it is a system_clock, it is not monotonic (e.g., the time can go backwards). For example, for gcc's libstdc++ it is system_clock, for MSVC it is steady_clock, and for clang's libc++ it depends on configuration.

**Review by jsteemann — APPROVED** [review]

**Comment by neunhoef** [comment]
This might need another fix for the unittests. Will check tomorrow. Do not merge for now.

**Comment by neunhoef** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13202/

**Comment by neunhoef** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13207/

**Comment by KVS85** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13405/

**Comment by KVS85** [comment]
[Set](https://github.com/arangodb/arangodb/pull/13184#event-4148954084) `DO_NOT_MERGE` label since https://github.com/arangodb/arangodb/pull/13057 was merged into this branch according to https://github.com/arangodb/arangodb/pull/13184#pullrequestreview-548547271 and had `DO_NOT_MERGE` label at the moment.

Review is required.

**Review by neunhoef — COMMENTED** [review]
LGTM

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13422/

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/13434/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13446/ new run after clang compiler bug was found.

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13448/ after making test more stable.

**Comment by KVS85** [comment]
Tests blue.

**Review by KVS85 — APPROVED** [review]
LGTM.

**Pull request merged**